Signal/src/ViewControllers/MediaPageViewController.swift

```diff
@@ -317,25 +317,51 @@ class MediaPageViewController: UIPageViewController, UIPageViewControllerDataSou
         let deleteAction = UIAlertAction(title: NSLocalizedString("TXT_DELETE_TITLE", comment: ""),
                                          style: .destructive) { _ in
                                             let deletedItem = currentViewController.galleryItem
-
-                                            if !self.sliderEnabled {
-                                                // In message details, which doesn't use the slider, so don't swap pages.
-                                            } else if let nextItem = mediaGalleryDataSource.galleryItem(after: deletedItem) {
-                                                self.setCurrentItem(nextItem, direction: .forward, animated: true)
-                                            } else if let previousItem = mediaGalleryDataSource.galleryItem(before: deletedItem) {
-                                                self.setCurrentItem(previousItem, direction: .reverse, animated: true)
-                                            } else {
-                                                // else we deleted the last piece of media, return to the conversation view
-                                                self.dismissSelf(animated: true)
-                                            }
-                                            mediaGalleryDataSource.delete(items: [deletedItem])
+                                            mediaGalleryDataSource.delete(items: [deletedItem], initiatedBy: self)
         }
         actionSheet.addAction(OWSAlerts.cancelAction)
         actionSheet.addAction(deleteAction)
 
         self.present(actionSheet, animated: true)
     }
 
+    // MARK: MediaGalleryDataSourceDelegate
+
+    func mediaGalleryDataSource(_ mediaGalleryDataSource: MediaGalleryDataSource, willDelete items: [MediaGalleryItem], initiatedBy: MediaGalleryDataSourceDelegate) {
```

Comment:
This is the crash fix - we need to remove pages referring to deleted items. This was already happening if the media was deleted in the page view, but not when deleted from the TileView.